app/models/work.rb

```diff
@@ -0,0 +1,49 @@
+class Work < ApplicationRecord
+  has_many :votes
+  has_many :users, through: :votes
+
+  validates :title, :creator, :category, presence: true
+  validates :title, uniqueness: { scope: :category }
+  validates :published, format: { with: /\d{4}/ }, numericality: { less_than_or_equal_to: Time.now.strftime("%Y").to_i }
+
+  def self.make_category_hash
+    categories = []
+    i = 0
+    while i < Work.all.length
+      if categories.include?(Work.all[i].category) == false
+        categories << Work.all[i].category
+      end
+      i += 1
+    end
+    return categories
+  end
+
+  def self.spotlight
+    if Work.all != []
```

Comment:
You should consider using an ennumerable like `max_by` or using ActiveRecord's `.order` method to do this.  

or just use `sort_by_vote` you made below.